internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
```

Comment:
since the function is trying to create cache directory as well, so I would prefer naming it like TryCreatingCacheDirectory and only return err. err should be nil if cache directory is created successfully and non-nil if not.

Reply:
done in latest commit